### PR TITLE
MIMIR-755: identify users with websocket sessions

### PR DESCRIPTION
**Pull request by itemodahl**
Pass user login to identify sessions
Add a user/session registry
Do websocket actions in user context
Remove outcommented logging

MIMIR-766

### CODE REVIEW

File: src/main/resources/lib/ssb/dataset/dashboard.ts [hunk 1]
```diff
@@ -18,31 +20,67 @@ const {
...
-      } else {
-        socketEmitter.broadcast('dashboard-activity-refreshDataset-result', {
-          id: id,
-          message: i18n.localize({
-            key: Events.FAILED_TO_FIND_DATAQUERY
-          }),
-          status: Events.FAILED_TO_FIND_DATAQUERY
-        })
+  socket.on('dashboard-register-user', (options: RegisterUserOptions) => {
+    log.info('update user')
```
Comment (Garlov) [inline src/main/resources/lib/ssb/dataset/dashboard.ts:35]: is all this logging needed?

File: src/main/resources/lib/ssb/dataset/dataset.ts [hunk 2]
```diff
@@ -6,13 +6,21 @@ import { JSONstat } from '../../types/jsonstat-toolkit'
 import { RepoQueryLib } from '../../repo/query'
 import { TbmlData } from '../../types/xmlParser'
 import { TbprocessorLib } from './tbprocessor'
+import {Context, ContextLibrary, RunContext} from "enonic-types/lib/context";
```
Comment (Garlov) [inline src/main/resources/lib/ssb/dataset/dataset.ts:9]: Context is not used
Linting on spacing and "

File: src/main/resources/lib/ssb/dataset/dataset.ts [hunk 3]
```diff
@@ -141,16 +169,19 @@ function isDataNew(data: JSONstat | TbmlData, dataset: DatasetRepoNode<JSONstat
 }
 
 export interface CreateOrUpdateStatus {
+  pewpew: number;
```
Comment (Garlov) [inline src/main/resources/lib/ssb/dataset/dataset.ts:172]: pewpew? 🔫

File: src/main/resources/services/websocket/websocket.es6 [hunk 4]
```diff
@@ -1,4 +1,6 @@
-const { run } = __non_webpack_require__('/lib/xp/context')
+const {
```
Comment (Garlov) [inline src/main/resources/services/websocket/websocket.es6:1]: run and auth is unused

File: src/main/resources/lib/convert/convert.ts [hunk 5]
```diff
@@ -210,10 +210,11 @@ function dataqueryToDataSource(dataquery: Content<Dataquery>): DataSource['dataS
 }
 
 export interface Socket {
```
Comment (Garlov) [inline src/main/resources/lib/convert/convert.ts:212]: Maybe move generic socket interfaces to "src\main\resources\lib\types"?

File: src/main/resources/lib/ssb/dataset/dashboard.ts [hunk 6]
```diff
@@ -7,6 +7,8 @@ import { getNode } from '../../repo/common'
 import { EVENT_LOG_REPO, EVENT_LOG_BRANCH } from '../../repo/eventLog'
 import { RepoNode } from 'enonic-types/lib/node'
 import { I18nLibrary } from 'enonic-types/lib/i18n'
+import { ContextLibrary, RunContext } from 'enonic-types/lib/context'
+import { AuthLibrary, User } from 'enonic-types/lib/auth'
```
Comment (Garlov) [inline src/main/resources/lib/ssb/dataset/dashboard.ts:11]: User is unused

File: src/main/resources/lib/ssb/dataset/dashboard.ts [hunk 7]
```diff
@@ -18,31 +20,67 @@ const {
   dateToFormat, dateToReadable
 } = __non_webpack_require__( '/lib/ssb/utils')
 const i18n: I18nLibrary = __non_webpack_require__('/lib/xp/i18n')
+const {
+  run
+}: ContextLibrary = __non_webpack_require__('/lib/xp/context')
+const {
+  getUser
```
Comment (Garlov) [inline src/main/resources/lib/ssb/dataset/dashboard.ts:27]: getUser is unused

File: src/main/resources/lib/ssb/dataset/dashboard.ts [hunk 8]
```diff
@@ -18,31 +20,67 @@ const {
...
-        socketEmitter.broadcast('dashboard-activity-refreshDataset-result', {
-          id: id,
-          message: i18n.localize({
-            key: Events.FAILED_TO_FIND_DATAQUERY
-          }),
-          status: Events.FAILED_TO_FIND_DATAQUERY
-        })
+  socket.on('dashboard-register-user', (options: RegisterUserOptions) => {
+    log.info('update user')
+    if(options && options.user) {
```
Comment (Garlov) [inline src/main/resources/lib/ssb/dataset/dashboard.ts:36]: spacing linting

File: src/main/resources/lib/ssb/dataset/dashboard.ts (hunk 1 above)
Comment (itemodahl) [reply src/main/resources/lib/ssb/dataset/dashboard.ts]: Nope. Removed.

File: src/main/resources/lib/ssb/dataset/dataset.ts (hunk 3 above)
Comment (itemodahl) [reply src/main/resources/lib/ssb/dataset/dataset.ts]: Fragments of testing ;) Removed

File: src/main/resources/lib/ssb/dataset/dataset.ts (hunk 2 above)
Comment (itemodahl) [reply src/main/resources/lib/ssb/dataset/dataset.ts]: Removed and linted

File: src/main/resources/services/websocket/websocket.es6 (hunk 4 above)
Comment (itemodahl) [reply src/main/resources/services/websocket/websocket.es6]: Removed

File: src/main/resources/lib/ssb/dataset/dashboard.ts (hunk 6 above)
Comment (itemodahl) [reply src/main/resources/lib/ssb/dataset/dashboard.ts]: Removed

File: src/main/resources/lib/ssb/dataset/dashboard.ts (hunk 8 above)
Comment (itemodahl) [reply src/main/resources/lib/ssb/dataset/dashboard.ts]: Linted

File: src/main/resources/lib/ssb/dataset/dashboard.ts (hunk 7 above)
Comment (itemodahl) [reply src/main/resources/lib/ssb/dataset/dashboard.ts]: Removed

File: src/main/resources/lib/convert/convert.ts (hunk 5 above)
Comment (itemodahl) [reply src/main/resources/lib/convert/convert.ts]: Moved to "src/main/resources/lib/types/socket.ts"?

**Review by Garlov — APPROVED** [review]
🥇

**Pull request merged**